staging/src/github.com/kubeedge/beehive/pkg/core/model/message.go

```diff
@@ -253,13 +253,20 @@ func (msg *Message) GetDestination() string {
 }
 
 // String the content that you want to send
-func (msg *Message) String() string {
+func (msg Message) String() string {
 	var buffer bytes.Buffer
 	buffer.WriteString("MessageID: " + msg.GetID())
-	buffer.WriteString(" ParentID: " + msg.GetParentID())
-	buffer.WriteString(" Source: " + msg.GetSource())
-	buffer.WriteString(" Destination: " + msg.GetDestination())
-	buffer.WriteString(" Resource: " + msg.GetResource())
-	buffer.WriteString(" Operation: " + msg.GetOperation())
+	buffer.WriteString(" ParentID: " + handleEmptyString(msg.GetParentID()))
+	buffer.WriteString(" Source: " + handleEmptyString(msg.GetSource()))
+	buffer.WriteString(" Destination: " + handleEmptyString(msg.GetDestination()))
+	buffer.WriteString(" Resource: " + handleEmptyString(msg.GetResource()))
+	buffer.WriteString(" Operation: " + handleEmptyString(msg.GetOperation()))
 	return buffer.String()
 }
+
+func handleEmptyString(s string) string {
+	if len(s) == 0 {
+		return "\"\""
```

Comment:
If the field is empty, I think we no need to set it to `""`, because the un-empty feild also has no the `""`.

And how about just printed it as json format:

```
buffer, _ := json.Marshal(msg)
return string(buffer)
```

Reply:
The purpose of setting an empty string to `""` is to make logging look better :)
Instead of this:
![image](https://user-images.githubusercontent.com/20016835/154021088-d75a9a58-9a36-4c52-9cb0-1d5d3da94591.png)
we can now get this:
![image](https://user-images.githubusercontent.com/20016835/154021307-3a13b318-503f-4f2c-984b-afdc30437faf.png)

---
Instead of using `json.Marshal`, we could use `fmt.Sprintf`:

```go
return fmt.Sprintf("MessageID: %s ParentID: %s Source: %s Destination: %s Resource: %s Operation: %s",
		msg.GetID(), msg.GetParentID(), msg.GetSource(), msg.GetDestination(), msg.GetResource(), msg.GetOperation())
```

Reply:
i guess we can just use %q to format string, this will add quotation marks to a string :)